src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -28,7 +29,8 @@ case class Instance(
     state: InstanceState,
     tasksMap: Map[Task.Id, Task],
     runSpec: RunSpec,
-    reservation: Option[Reservation]) extends Placed {
+    reservation: Option[Reservation],
+    role: Role) extends Placed {
```

Comment:
Could you add a comment to the instance? Eg

```scala
/**
 * Internal state of a instantiated [[RunSpec]], ie instance of an [[AppDefinition]]
 * or [[PodDefinition]].
 * ...
 * @param role The Mesos role the instance was launched this. It can differ
 *             from the run spec role. When a resident instance is updated and
 *             restarted we keep the old role to maintain the data. For instance
 *             one might want to update the Cassandra nodes from v1 to v2 but
 *             loose terabytes of data. All old instances would keep running with
 *             role `slave_public` even though the app definition for v2 as role
 *             `prod` and all new instance are launched with role `pod`. This
 *             inconsistency will allow for a smooth migration of persistent apps.
 */
case class Instance(...
    role: Role
```

Reply:
(Karsten, I hope you don't mind, I edited your comment and re-worded the comment about the instance parameter, shooting for more brevity)